Dockerfile

```diff
@@ -0,0 +1,166 @@
+ARG unitycatalog_uid=185
+ARG unitycatalog_home="/opt/unitycatalog"
+ARG unitycatalog_repo="${unitycatalog_home}/repo"
+ARG unitycatalog_jar="server/target"
+ARG unitycatalog_etc="etc"
+ARG unitycatalog_bin="bin"
+ARG unitycatalog_user_name="unitycatalog"
+ARG unitycatalog_user_home="home"
+ARG unitycatalog_user_basedir="${unitycatalog_home}/${unitycatalog_user_home}"
+# Specify any custom parameters necessary to generate
+# the Uber-Jar by SBT. 
+# Note: The default allocated heam memory size is too small
+# and will cause the process to fail when attempting to compile
+# and generate the Uber-Jar. Therefore it is important to choose
+# a size large enough for the compiler to run. 
+ARG sbt_args="-J-Xmx5G"
+
+
+
+# ###### STAGE 1 ###### #
+# Building the Uber-Jar #
+#########################
+
+FROM eclipse-temurin:22-jdk-alpine AS build_uc
+
+ARG unitycatalog_repo
+ARG sbt_args
+
+# Install any required packages
+RUN <<EOF
+    set -ex;
+    apk update;
+    apk upgrade; 
+    apk add bash git;
+    rm -R /var/cache/apk/*;
+EOF
+
+# Create temporary directories for initialization files
+RUN mkdir -p "${unitycatalog_repo}";
+
+WORKDIR "${unitycatalog_repo}"
+
+COPY . .
+
+RUN build/sbt ${sbt_args} server/assembly
+
+
+# ###### STAGE 2 ###### #
+# Running the UC server #
+#########################
+
+FROM eclipse-temurin:22-jre-alpine AS run_uc
+
+ARG unitycatalog_uid
+ARG unitycatalog_home
+ARG unitycatalog_repo
+ARG unitycatalog_jar
+ARG unitycatalog_etc
+ARG unitycatalog_bin
+ARG unitycatalog_user_name
+ARG unitycatalog_user_home
+ARG unitycatalog_user_basedir
+ARG sbt_args
+
+EXPOSE 8081
+
+RUN <<EOF
+    set -ex;
+    apk update;
+    apk upgrade; 
+    apk add bash;
+    rm -R /var/cache/apk/*;
+EOF
+
+# Define the shell used within the container
+SHELL ["/bin/bash", "-i", "-c", "-o", "pipefail"] 
+
+ENV SERVER_TARGET_DIR="${unitycatalog_home}/${unitycatalog_jar}"
+ENV UC_SERVER_BIN="${unitycatalog_home}/${unitycatalog_bin}/start-uc-server"
+
+# Create the UC directories
+RUN <<-EOF 
+    set -ex;
+    mkdir -p "${unitycatalog_home}";
+    mkdir -p "${unitycatalog_home}/${unitycatalog_jar}";
+    mkdir -p "${unitycatalog_home}/${unitycatalog_etc}";
+    mkdir -p "${unitycatalog_home}/${unitycatalog_bin}";
+    mkdir -p "${unitycatalog_home}/${unitycatalog_user_home}";
```

Comment:
I'm curious as to why don't we use the tarball we generate directly? that would save us: stage 1, additional setting up of these dirs and a bunch of code i presume?

Reply:
When we started this PR the tarball was not yet created :laughing:

Reply:
I will look into it this weekend and come back either with a change or a suggestion :wink: